The job **`x86_64-gnu-llvm-15`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/6263440819/job/17007849844) [(plain)](https://github.com/rust-lang/rust/commit/548ae45e40d6852f949d7bf5aa9a90250e8cb0a5/checks/17007849844/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
GITHUB_ACTION=__run_7
GITHUB_ACTIONS=true
GITHUB_ACTION_REF=
GITHUB_ACTION_REPOSITORY=
GITHUB_ACTOR=chenyukang
GITHUB_API_URL=https://api.github.com
GITHUB_BASE_REF=master
GITHUB_ENV=/home/runner/work/_temp/_runner_file_commands/set_env_be02e4ff-0d8c-4b5d-bff6-d905b97668d3
GITHUB_EVENT_NAME=pull_request
GITHUB_EVENT_NAME=pull_request
GITHUB_EVENT_PATH=/home/runner/work/_temp/_github_workflow/event.json
GITHUB_GRAPHQL_URL=https://api.github.com/graphql
GITHUB_HEAD_REF=yukang-98601-add-ui-testcase
GITHUB_JOB=pr
GITHUB_PATH=/home/runner/work/_temp/_runner_file_commands/add_path_be02e4ff-0d8c-4b5d-bff6-d905b97668d3
GITHUB_REF=refs/pull/116034/merge
GITHUB_REF_NAME=116034/merge
GITHUB_REF_PROTECTED=false
---
GITHUB_SERVER_URL=https://github.com
GITHUB_SHA=3b8495aac63fd090454900c09254884c5a386488
GITHUB_STATE=/home/runner/work/_temp/_runner_file_commands/save_state_be02e4ff-0d8c-4b5d-bff6-d905b97668d3
GITHUB_STEP_SUMMARY=/home/runner/work/_temp/_runner_file_commands/step_summary_be02e4ff-0d8c-4b5d-bff6-d905b97668d3
GITHUB_TRIGGERING_ACTOR=chenyukang
GITHUB_WORKFLOW_REF=rust-lang/rust/.github/workflows/ci.yml@refs/pull/116034/merge
GITHUB_WORKFLOW_SHA=3b8495aac63fd090454900c09254884c5a386488
GITHUB_WORKSPACE=/home/runner/work/rust/rust
GOROOT_1_18_X64=/opt/hostedtoolcache/go/1.18.10/x64
---
..................i..................................................................... 10208/15603
.......i................................................................................ 10296/15603
........................................................................................ 10384/15603
........................................................i............................... 10472/15603
..............................................................................F.F....... 10560/15603
........................................................................................ 10736/15603
........................................................................................ 10824/15603
........................................................................................ 10912/15603
........................................................................................ 11000/15603
---
To only update this specific test, also pass `--test-args parser/issues/issue-98601-delimiter-error-unexpected-close.rs`

error: 1 errors occurred comparing output.
status: exit status: 1
command: RUSTC_ICE="0" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/tests/ui/parser/issues/issue-98601-delimiter-error-unexpected-close.rs" "-Zthreads=1" "-Zsimulate-remapped-rust-src-base=/rustc/FAKE_PREFIX" "-Ztranslate-remapped-path-to-local-path=no" "-Z" "ignore-directory-in-diagnostics-source-blocks=/cargo" "--sysroot" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2" "--target=x86_64-unknown-linux-gnu" "--error-format" "json" "--json" "future-incompat" "-Ccodegen-units=1" "-Zui-testing" "-Zdeduplicate-diagnostics=no" "-Zwrite-long-types-to-disk=no" "-Cstrip=debuginfo" "--emit" "metadata" "-C" "prefer-dynamic" "--out-dir" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/parser/issues/issue-98601-delimiter-error-unexpected-close" "-A" "unused" "-A" "internal_features" "-Crpath" "-Cdebuginfo=0" "-Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/parser/issues/issue-98601-delimiter-error-unexpected-close/auxiliary"
--- stderr -------------------------------
error: unexpected closing delimiter: `)`
##[error]  --> /checkout/tests/ui/parser/issues/issue-98601-delimiter-error-unexpected-close.rs:5:17
   |
   |
LL | fn main() {
   |           - this opening brace...
LL |     todo!();
LL | }
   | - ...matches this closing brace
LL |
LL | fn other(_: i32)) {} //~ ERROR unexpected closing delimiter: `)`

error: aborting due to previous error
------------------------------------------

---
To only update this specific test, also pass `--test-args parser/issues/issue-98601-delimiter-error-1.rs`

error: 1 errors occurred comparing output.
status: exit status: 1
command: RUSTC_ICE="0" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/tests/ui/parser/issues/issue-98601-delimiter-error-1.rs" "-Zthreads=1" "-Zsimulate-remapped-rust-src-base=/rustc/FAKE_PREFIX" "-Ztranslate-remapped-path-to-local-path=no" "-Z" "ignore-directory-in-diagnostics-source-blocks=/cargo" "--sysroot" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2" "--target=x86_64-unknown-linux-gnu" "--error-format" "json" "--json" "future-incompat" "-Ccodegen-units=1" "-Zui-testing" "-Zdeduplicate-diagnostics=no" "-Zwrite-long-types-to-disk=no" "-Cstrip=debuginfo" "--emit" "metadata" "-C" "prefer-dynamic" "--out-dir" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/parser/issues/issue-98601-delimiter-error-1" "-A" "unused" "-A" "internal_features" "-Crpath" "-Cdebuginfo=0" "-Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/parser/issues/issue-98601-delimiter-error-1/auxiliary"
--- stderr -------------------------------
error: unexpected closing delimiter: `}`
##[error]  --> /checkout/tests/ui/parser/issues/issue-98601-delimiter-error-1.rs:7:1
   |
   |
LL | fn foo() {
   |          - this delimiter might not be properly closed...
LL |     match 0 {
LL |       _ => {}
   |            -- block is empty, you might have not meant to close it
LL |     }
LL |     }
   |     - ...as it matches this but it has different indentation
LL | } //~ ERROR unexpected closing delimiter: `}`

error: aborting due to previous error
------------------------------------------

```

</details>

--- a/tests/ui/parser/issues/issue-98601-delimiter-error-1.rs
+++ b/tests/ui/parser/issues/issue-98601-delimiter-error-1.rs
@@ -1,0 +1,9 @@
+fn foo() {
+    match 0 {
+      _ => {}
+    }
+    if foo
+    }
+} //~ ERROR unexpected closing delimiter: `}`
+
+fn main() {}

--- a/tests/ui/parser/issues/issue-98601-delimiter-error-1.stderr
+++ b/tests/ui/parser/issues/issue-98601-delimiter-error-1.stderr
@@ -1,0 +1,16 @@
+error: unexpected closing delimiter: `}`
+  --> $DIR/issue-98601-delimiter-error-1.rs:7:1
+   |
+LL | fn foo() {
+   |          - this delimiter might not be properly closed...
+LL |     match 0 {
+LL |       _ => {}
+   |            -- block is empty, you might have not meant to close it
+...
+LL |     }
+   |     - ...as it matches this but it has different indentation
+LL | }
+   | ^ unexpected closing delimiter
+
+error: aborting due to previous error
+

--- a/tests/ui/parser/issues/issue-98601-delimiter-error-unexpected-close.rs
+++ b/tests/ui/parser/issues/issue-98601-delimiter-error-unexpected-close.rs
@@ -1,0 +1,5 @@
+fn main() {
+    todo!();
+}
+
+fn other(_: i32)) {} //~ ERROR unexpected closing delimiter: `)`

--- a/tests/ui/parser/issues/issue-98601-delimiter-error-unexpected-close.stderr
+++ b/tests/ui/parser/issues/issue-98601-delimiter-error-unexpected-close.stderr
@@ -1,0 +1,14 @@
+error: unexpected closing delimiter: `)`
+  --> $DIR/issue-98601-delimiter-error-unexpected-close.rs:5:17
+   |
+LL | fn main() {
+   |           - this opening brace...
+LL |     todo!();
+LL | }
+   | - ...matches this closing brace
+LL |
+LL | fn other(_: i32)) {}
+   |                 ^ unexpected closing delimiter
+
+error: aborting due to previous error
+

